lib/actions/sessions.js

```diff
@@ -53,22 +53,23 @@ export function addSessionData(uid, data) {
       type: SESSION_ADD_DATA,
       data,
       effect() {
-        const {shell} = getState().sessions.sessions[uid];
-        const enterKey = data.indexOf('\n') > 0;
-        const url = enterKey ? isUrl(shell, data) : null;
-        if (url) {
-          dispatch({
-            type: SESSION_URL_SET,
-            uid,
-            url
-          });
-        } else {
-          dispatch({
-            type: SESSION_PTY_DATA,
-            uid,
-            data
-          });
+        const session = getState().sessions.sessions[uid];
+        if (session) {
+          const enterKey = data.indexOf('\n') > 0;
+          const url = enterKey ? isUrl(session.shell, data) : null;
+          if (url) {
+            return dispatch({
+              type: SESSION_URL_SET,
+              uid,
+              url
+            });
+          }
         }
+        dispatch({
+          type: SESSION_PTY_DATA,
+          uid,
+          data
+        });
```

Comment:
Why has this been moved outside of the else?  Will it happen twice if it isn't a URL?